specification/network/resource-manager/Microsoft.Network/stable/2024-01-01/bastionHost.json

```diff
@@ -640,6 +640,121 @@
           }
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/bastionHosts/{bastionHostName}/setSessionRecordingSasUrl": {
```

Comment:
Calling your API 'setSessionRecordingSasUrl' makes it extremely implementation detail specific. What about a more neutral name: 'setSessionRecordingSettings'?

Also, have you thought about the option of making it a PUT API instead of a POST API? I don't understand yet why you went with POST.

If those settings are a singleton it could have been :
PUT .../sessionRecordingSettings/default
and then you get updates PUT or PATCH (or DELETE?) to the same path, and GET for reads.

versus 
POST .../setSessionRecordingSettings
for read/write
which ends up being a whole new set of RBAC actions (did you want or need special RBAC actions? Why?)

Reply:
POST is used because the call needs to reach Bastion tenant thereby making it a long running operation. And it's only used to set the SAS URL while nothing else is needed hence the name

Reply:
Please can you explain why POST reaches Bastion tenant and PUT wont?

Reply:
@madhura-np Since this is not an update of Bastion resource itself, we have chosen to go with POST

Reply:
I'm OK with this being a POST, primarily because the payload is a secret that is not stored in the resource. Since querying secrets requires separate RBAC, that puts it in the POST realm. Per other comments, the payload should be changed to JSON, which implies a slightly more generic name for the action is warranted as Tim suggested.

Reply:
@mentat9 Payload for 'Set' is a json. As for the generic naming, since this API will only be used for setting SAS URL, this should be good for now